kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -187,13 +176,26 @@ class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase
     }
   }
 
+  override def commitTransaction(
+      txn: Transaction,
+      engine: Engine,
+      dataActions: CloseableIterable[Row]): TransactionCommitResult = {
+    val result = txn.commit(engine, dataActions)
+    result.getPostCommitHooks
+      .stream()
+      .forEach(hook => hook.threadSafeInvoke(engine))
+    result
+  }
```

Comment:
should we make this a mixin trait maybe that has this? can be a follow-up refactor PR

Reply:
will address as a follow up